sys/fs/cd9660/cd9660_node.c

```diff
@@ -146,8 +146,22 @@ cd9660_defattr(struct iso_directory_record *isodir, struct iso_node *inop,
 	}
 	if (!ap) {
 		inop->inode.iso_mode |= S_IRUSR|S_IXUSR|S_IRGRP|S_IXGRP|S_IROTH|S_IXOTH;
-		inop->inode.iso_uid = (uid_t)0;
-		inop->inode.iso_gid = (gid_t)0;
+		/*
+		 * Set some "sane" permissions for non-Rockridge volumes
+		 */
+		switch (inop->inode.iso_mode & S_IFMT) {
+		case S_IFREG:
+			/* Turn off executable bits for regular files */
+			inop->inode.iso_mode &= ~(VEXEC | VEXEC >> 3 | VEXEC >> 6);
```

Comment:
I would suggest using `S_IX*` instead of `VEXEC` with shifts.  However, I'm not sure if makes sense to turn off executable permissions on all files as a non-rock-ridge image may contain executables.

Reply:
I agree with the `S_IX*` part but turning off the executable bits means a sane default for me as I wouldn't expect executables from non-Rock-Ridge.

Reply:
I removed the whole change of this block.